q2_feature_table/_normalize.py

```diff
@@ -5,17 +5,26 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
-
+import random
 import biom
 
 
-def rarefy(table: biom.Table, sampling_depth: int,
-           with_replacement: bool = False) -> biom.Table:
+def rarefy(table: biom.Table,
+           sampling_depth: int,
+           with_replacement: bool = False,
+           random_seed: float = None
+           ) -> biom.Table:
+    # Generate a random seed if seed is None
+    if random_seed is not None:
+        random.seed(random_seed)
```

Comment:
I think this should get you what you are looking for:

```suggestion
    random.seed(a=random_seed)
```

From [the docs](https://docs.python.org/3/library/random.html#random.seed):

> If a is omitted or None, the current system time is used. If randomness sources are provided by the operating system, they are used instead of the system time (see the os.urandom() function for details on availability).
> If a is an int, it is used directly.